Sources/TuistGenerator/Linter/SchemeLinter.swift

```diff
@@ -79,4 +80,33 @@ private extension SchemeLinter {
         let reason = "The target '\(missingTargetName)' specified in \(schemaName) code coverage targets list isn't defined in the project."
         return LintingIssue(reason: reason, severity: .error)
     }
+    
+    func projectSchemeCantReferenceRemoteTargets(schemes: [Scheme], project: Project) -> [LintingIssue] {
+        return schemes.flatMap { projectSchemeCantReferenceRemoteTargets(scheme: $0, project: project) }
+    }
+    
+    func projectSchemeCantReferenceRemoteTargets(scheme: Scheme, project: Project) -> [LintingIssue] {
+        var issues: [LintingIssue] = []
+        var targets = [TargetReference?]()
+        
+        scheme.buildAction?.targets.forEach { targets.append($0) }
```

Comment:
What about having a method in the `Scheme` model that returns all the targets the scheme depends on?: `scheme.targetDependencies()`? That way we could unit-test it easily.